R/apa_factorial_plot.R

```diff
@@ -277,11 +281,15 @@ apa_factorial_plot.default <- function(
   if(fun_dispersion == "within_subjects_conf_int" || fun_dispersion == "wsci") {
     ee <- wsci(data = aggregated, id = id, factors = factors, level = level, method = "Morey", dv = dv)
   } else {
-    if(fun_dispersion == "conf_int") {
-      ee <- stats::aggregate(formula = stats::as.formula(paste0(dv, "~", paste(factors, collapse = "*"))), data = aggregated, FUN = dispersion, level = level)
-    } else {
-      if(use_dplyr) {
+    if(use_dplyr) {
+      if(fun_dispersion=="conf_int"){
+        ee <- fast_aggregate(data = aggregated, factors = factors, dv = dv, fun = dispersion, level = level)
```

Comment:
Could we make this more generic so that we add an option to pass arguments to the dispersion function (e.g., `args_dispersion`)? That would make it possible to choose a method for `wsci()`, for example. Another application  I can think off would be to pass a dispersion function that computes bootstrap confidence intervals and would take options such as the number of bootstrap samples.